Formatter/Pool.php

```diff
@@ -29,20 +31,60 @@ class Pool
     protected $defaultFormatter;
 
     /**
+     * NEXT_MAJOR: use LoggerAwareTrait.
+     *
      * @var LoggerInterface
      */
     protected $logger;
 
     /**
-     * @param LoggerInterface|null $logger
-     * @param string|null          $defaultFormatter
+     * @param string|null $defaultFormatter
      */
-    public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
+    public function __construct($defaultFormatter = null)
     {
-        $this->logger = $logger;
+        if (func_num_args() == 2) {
+            list($logger, $defaultFormatter) = func_get_args();
+            $this->logger = $logger;
+            $this->defaultFormatter = $defaultFormatter;
+        } elseif (func_num_args() == 1) {
```

Comment:
@sagikazarmark : I always said the Jedi council has too much influence :) [Look how carried away they can get](https://github.com/symfony/symfony-docs/pull/5381)